startup_scripts/250_dcim_interfaces.py

```diff
@@ -0,0 +1,55 @@
+from dcim.models import Interface, Device
+from dcim.constants import IFACE_TYPE_CHOICES
+from extras.models import CustomField, CustomFieldValue
+from ruamel.yaml import YAML
+
+from pathlib import Path
+import sys
+
+file = Path('/opt/netbox/initializers/dcim_interfaces.yml')
+if not file.is_file():
+  sys.exit()
+
+with file.open('r') as stream:
+  yaml = YAML(typ='safe')
+  interfaces = yaml.load(stream)
+
+  required_assocs = {
+    'device': (Device, 'name')
+  }
+
+  if interfaces is not None:
+    for params in interfaces:
+      custom_fields = params.pop('custom_fields', None)
+
+      for assoc, details in required_assocs.items():
+        model, field = details
+        query = { field: params.pop(assoc) }
+
+        params[assoc] = model.objects.get(**query)
+
+      if 'form_factor' in params:
+        for outer_list in IFACE_TYPE_CHOICES:
+          for ffactor_choices in outer_list[1]:
+            if params['form_factor'] in ffactor_choices:
+              params['form_factor'] = ffactor_choices[0]
+              break
+          else:
+            continue
+          break
```

Comment:
This `break` is not guarded by an `if`, so it's always executed. This looks like an error to me.
If that the intended behavior then I the `for` loop on line 32 should probably be an `if` statement.

Reply:
IFACE_TYPE_CHOICES is a list of lists, and a relatively long one, I needed a way to break out of the 2 loops when the value was found.
If the inner loop (for ffactor_choices in outer_list[1]:) breaks, then the "else" clause is not executed and the "break" in line 39 does get executed, thus breaking the outer loop.
If the inner loop does not break, the else clause gets executed which skips the execution of "break" in line 39.

Reply:
IFACE_TYPE_CHOICES is a list of lists, and a relatively long one, I needed a way to break out of the 2 loops when the value was found.
If the inner loop (for ffactor_choices in outer_list[1]:) breaks, then the "else" clause is not executed and the "break" in line 39 does get executed, thus breaking the outer loop.
If the inner loop does not break, the else clause gets executed which skips the execution of "break" in line 39.